integration/app_integration_test.go

```diff
@@ -1565,26 +1571,29 @@ func (p *pack) startRootAppServers(t *testing.T, count int, extraApps []service.
 			},
 		}, extraApps...)
 
-		appServer, err := p.rootCluster.StartApp(raConf)
-		require.NoError(t, err)
-		t.Cleanup(func() { appServer.Close() })
+		configs[i] = raConf
+	}
+
+	servers, err := p.rootCluster.StartApps(configs)
+	require.NoError(t, err)
 
-		servers[i] = appServer
+	for _, appServer := range servers {
+		t.Cleanup(func() { require.NoError(t, appServer.Close()) })
 	}
 
 	return servers
 }
 
 func (p *pack) startLeafAppServers(t *testing.T, count int, extraApps []service.App) []*service.TeleportProcess {
 	log := utils.NewLoggerForTests()
-	servers := make([]*service.TeleportProcess, count)
+	configs := make([]*service.Config, count)
 
 	for i := 0; i < count; i++ {
 		laConf := service.MakeDefaultConfig()
 		laConf.Console = nil
 		laConf.Log = log
 		laConf.DataDir = t.TempDir()
-		t.Cleanup(func() { os.RemoveAll(laConf.DataDir) })
+		t.Cleanup(func() { require.NoError(t, os.RemoveAll(laConf.DataDir)) })
```

Comment:
If we use `t.TempDir()` then there's no need for `os.RemoveAll()`, right?

Reply:
Yeah, good call. It would be nice if we moved `TeleInstance` to use `t.TempDir` for all things